src/core/gui/dialog/LatexDialog.cpp

```diff
@@ -86,6 +243,69 @@ auto LatexDialog::getBufferContents() -> std::string {
     return s;
 }
 
+void LatexDialog::drawPreviewCallback(GtkWidget* widget, cairo_t* cr, LatexDialog* self) {
+    GtkStyleContext* context = gtk_widget_get_style_context(widget);
+
+    guint widgetWidth = gtk_widget_get_allocated_width(widget);
+    guint widgetHeight = gtk_widget_get_allocated_height(widget);
+
+    gtk_render_background(context, cr, 0, 0, widgetWidth, widgetHeight);
+
+    // If we have nothing to render, do nothing!
+    if (!self->tempRenderSource) {
+        return;
+    }
+
+    double scaleFactor = self->getPreviewScale();
+
+    // If possible, try not to make the preview larger than rendered.
+    if (scaleFactor > 1.0) {
+        self->renderScaledPreview();
+
+        // Don't increase the size of the preview if we're at the maximum scale.
+        scaleFactor = std::max(1.0, self->getPreviewScale());
+    }
+
+    int renderedWidth = cairo_image_surface_get_width(self->scaledRender);
+    int renderedHeight = cairo_image_surface_get_height(self->scaledRender);
+
+    // Don't render empty previews.
+    if (renderedWidth == 0 || renderedHeight == 0) {
+        return;
+    }
+
+    // Center the image horizontally and vertically.
+    double translateX = 0.0, translateY = 0.0;
+
+    double extraHorizontalSpace = std::max(0.0, widgetWidth - renderedWidth * scaleFactor);
+    double extraVerticalSpace = std::max(0.0, widgetHeight - renderedHeight * scaleFactor);
+
+    translateX = extraHorizontalSpace / 2.0;
+    translateY = extraVerticalSpace / 2.0;
+
+    cairo_matrix_t matOriginal;
+    cairo_matrix_t matTransformed;
+    cairo_get_matrix(cr, &matOriginal);
+    cairo_get_matrix(cr, &matTransformed);
+
+    matTransformed.xx = scaleFactor;
+    matTransformed.yy = scaleFactor;
+    matTransformed.xy = 0;
+    matTransformed.yx = 0;
+    matTransformed.x0 += translateX;
+    matTransformed.y0 += translateY;
```

Comment:
That could also be written as follows:
```suggestion
    cairo_matrix_init(&matTransformed, scaleFactor, 0, 0, scaleFactor, matOriginal.x0 + translateX, matOriginal.y0 + translateY);
```

Reply:
As above, I personally find the longer/multi-line form easier to read. Of course, if you disagree, I'm willing to change it!

Reply:
Those two were only suggestions, a matter of taste. I don't mind if you keep it the way it is.